### PR TITLE
Neural Operator Implementation

### PR DESCRIPTION
**Pull request by deepcodin**

### CODE REVIEW

File: src/dynabench/model/_grid/_neural_operator.py [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+        # Fourier Blocks
+        layers = [NeuralFourierBlock2d(width, n_modes, activation=True) for i in range(n_layers - 1)]
+        layers.append(NeuralFourierBlock2d(width, n_modes, activation=False))
+        self.fourier_blocks = nn.Sequential(*layers)
+
+        # Reduce to output dimension
+        self.fc1 = nn.Linear(width, 128)
+        self.fc2 = nn.Linear(128, channels)
+
+    def forward(self, x: torch.Tensor, p: torch.Tensor):
```
Comment (badulion) [inline src/dynabench/model/_grid/_neural_operator.py:159]: p is not used in the forward pass -> should not be passed as an argument

File: examples/example_neural_operator.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+optimizer = optim.Adam(model.parameters(), lr=1e-3)
+criterion = nn.MSELoss()
+
+for epoch in range(1):
+    model.train()
+    for i, (x, y, p) in enumerate(train_loader):
+        x, y, p = x[:,0].float(), y.float(), p.float() # only use the first channel and convert to float32
+        optimizer.zero_grad()
+        y_pred = model(x, p)
```
Comment (badulion) [inline examples/example_neural_operator.py:32]: p is not required for grid models (see e.g. NeuralPDE)

File: examples/example_neural_operator.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
+from dynabench.dataset import DynabenchIterator, download_equation
+from torch.utils.data import DataLoader
+from dynabench.model._grid._neural_operator import FourierNeuralOperator
+from dynabench.model.utils import PointIterativeWrapper
```
Comment (badulion) [inline examples/example_neural_operator.py:4]: A new IterativeWrapper for grid models needs to be created, which does not require the positions as input

**Review by badulion — CHANGES_REQUESTED** [review]
Summary:
- do not use positions p for grid models
- create new iterative wrapper for grid model iterative predictions

File: examples/example_neural_operator.py (hunk 3 above)
Comment (deepcodin) [reply examples/example_neural_operator.py]: implemented grid wrapper in model/utils.py and used for neural operator -> points are no longer passed to forward()

**Comment by deepcodin** [comment]
All changes implemented.

**Review by badulion — APPROVED** [review]

**Pull request merged**